src/alt.js

```diff
@@ -460,7 +460,7 @@ class Alt {
   takeSnapshot(...storeNames) {
     const state = snapshot(this, ...storeNames)
     if (this[LAST_SNAPSHOT]) {
-      assign(this[LAST_SNAPSHOT], state)
+      this[LAST_SNAPSHOT] = JSON.stringify(assign(JSON.parse(this[LAST_SNAPSHOT]), JSON.parse(state)))
```

Comment:
I like the second one, return obj from snapshot and stringify at flush and takesnapshot. You'll still have to JSON parse the last snapshot though.